core/src/com/unciv/logic/city/CityInfo.kt

```diff
@@ -344,7 +344,20 @@ class CityInfo {
                 if (!matchesFilter(unique.params[1])) continue
                 stats[entry.key]!!.timesInPlace(1 + unique.params[0].toFloat() / 100f)
             }
-            
+
+            // Sweden UP
+            var friendshipMultiplier = 0f
+            for (otherciv in civInfo.getKnownCivs()) {
+                if (civInfo.getDiplomacyManager(otherciv).hasFlag(DiplomacyFlags.DeclarationOfFriendship)) {
+                    for(ourunique in civInfo.getMatchingUniques("When declaring [], both parties gain a +[]% boost to []"))
```

Comment:
- `friendship` does not need to be parametrized, as it cannot be filled with any other value. 
- The `+` sign should be inside the parameterization, to allow a `-` sign in some situations
- Do you keep track of the different amounts for the different stats? From what you have written it looks like you multiply all stats equally?

Reply:
I'm unsure what you mean by the last point? In this loop the only entries should be GP points from buildings and GP points from specialists, correct? In that case they should be multiplied equally.
Also probably [great people generation] makes little sense as a parameter as I didn't code an implementation for science or whatever

Reply:
I misunderstood your code, for some reason I thought it would be used for `[amount]% boost to [Science]` or something instead of [great people generation]. It does make sense now that I look at it again.

Reply:
I mean, it could, but I didn't implement that

Reply:
Having thought about it I simply removed the parameter for [great people generation] so it reflects the implementation